tests/std/tests/P0528R3_cmpxchg_pad/test.cpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <atomic>
+#include <cassert>
+#include <cstddef>
+#include <cstring>
+
+struct X1 {
+    char x : 6;
+
+    void set(char v) {
+        x = v;
+    }
+};
+
+struct X2 {
+    short x : 9;
+
+    void set(char v) {
+        x = v;
+    }
+};
+
+#pragma pack(push, 1)
+struct X3 {
+    char x : 4;
+    char : 2;
+    char y : 1;
+    short z;
+
+    void set(char v) {
+        x = v;
+        y = 0;
+        z = 0;
+    }
+};
+#pragma pack(pop)
+
+#pragma warning(push)
+#pragma warning(disable : 4324) // '%s': structure was padded due to alignment specifier
+struct alignas(4) X4 {
+    char x;
+
+    void set(char v) {
+        x = v;
+    }
+};
+#pragma warning(pop)
+
+struct X8 {
+    char x;
+    long y;
+
+    void operator&() const {}
+
+    void set(char v) {
+        x = v;
+        y = 0;
+    }
+};
+
+struct X16 {
+    long x;
+    char y;
+    long long z;
+
+    void set(char v) {
+        x = v;
+        y = 0;
+        z = 0;
+    }
+};
+
+struct X20 {
+    long x;
+    long y[3];
+    char z;
+
+    void set(char v) {
+        x = v;
+        std::memset(&y, 0, sizeof(y));
+        z = 0;
+    }
+};
+
+
+template <class X, std::size_t S>
+void test() {
+    static_assert(sizeof(X) == S, "Unexpected size");
+    X x2;
+    X x3;
+    X x4;
+    X x1;
+    std::memset(std::addressof(x1), 0xaa, sizeof(x1));
+    std::memset(std::addressof(x2), 0x55, sizeof(x1));
+    std::memset(std::addressof(x3), 0x55, sizeof(x1));
+    x1.set(5);
+    x2.set(5);
+    x3.set(6);
+    x4.set(7);

```

Comment:
Is it intentional that `x4` isn't `std::memset`, only `.set`? If so, should this be commented?

Reply:
no